src/ledger/LedgerTxn.h

```diff
@@ -449,6 +449,10 @@ class AbstractLedgerTxnParent
     // work, while still being correct. Will throw when called on anything other
     // than a (real or stub) root LedgerTxn.
     virtual uint32_t prefetch(UnorderedSet<LedgerKey> const& keys) = 0;
+
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+    virtual void resetForFuzzer() = 0;
```

Comment:
The fact that you need to define a function in three classes that is implemented as `abort()` in all of them suggests that this is not a good design. How about we inject a class `AbstractLedgerTxnRoot` in the hierarchy between `AbstractLedgerTxnParent` and `LedgerTxnRoot`? This would be make it `AbstractLedgerTxnParent, AbstractLedgerTxnRoot, LedgerTxnRoot` which is analogous to the existing hierarchy is `AbstractLedgerTxnParent, AbstractLedgerTxn, LedgerTxn`.

This would actually solve some other problems, notably that the `LedgerTxn::drop*` functions always throw so they probably shouldn't exist at all using my same logic as above. 

I don't think this needs to be done in this PR. If you elect not to do it, please open an issue so that we don't forget.

Reply:
I don't think I'd want to add new abstractions just to make the fuzzing code happy - if you see other use cases for that yeah we could do it at some point.